### PR TITLE
[compiler-rt][profile] Add support for LLVM profile for Haiku

### PR DESCRIPTION
**Pull request by korli**
Haiku uses typical UNIX interfaces.

All tests pass except
instrprof-error.c
Posix/gcov-dlopen.c
Posix/gcov-destructor.c
Posix/instrprof-dlopen-norpath.test

### CODE REVIEW

**Comment by brad0** [comment]
@MaskRay

**Comment by brad0** [comment]
cc @MaskRay

File: compiler-rt/cmake/config-ix.cmake [hunk 1]
```diff
@@ -37,7 +37,11 @@ check_c_compiler_flag(-nodefaultlibs C_SUPPORTS_NODEFAULTLIBS_FLAG)
 if (C_SUPPORTS_NODEFAULTLIBS_FLAG)
   set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -nodefaultlibs")
   if (COMPILER_RT_HAS_LIBC)
-    list(APPEND CMAKE_REQUIRED_LIBRARIES c)
+    if (HAIKU)
+      list(APPEND CMAKE_REQUIRED_LIBRARIES root)
+    else ()
+      list(APPEND CMAKE_REQUIRED_LIBRARIES c)
+    endif ()
```
Comment (petrhosek) [inline compiler-rt/cmake/config-ix.cmake:44]:
```suggestion
    if (HAIKU)
      list(APPEND CMAKE_REQUIRED_LIBRARIES root)
    else()
      list(APPEND CMAKE_REQUIRED_LIBRARIES c)
    endif()
```

**Review by petrhosek — APPROVED** [review]

**Comment by brad0** [comment]
@korli Please rebase.

**Comment by korli** [comment]
> @korli Please rebase.

Done

**Pull request merged**